Cargo.toml

```diff
@@ -140,3 +140,10 @@ paste = "1.0"
 rand = "0.8.5"
 time = "0.3"
 trait-set = "0.3.0"
+
+[profile.dev]
```

Comment:
Would welcome comments on why these values are set the way they are.

I think the last one does opt-level 3 for dependencies, which makes sense because we don't have to recompile them unless we change rust version or RUSTFLAGS. But I'm not sure why the others are set to their current values.

Reply:
Also wondering if we want to set the `test` profile instead: https://doc.rust-lang.org/cargo/reference/profiles.html#test

Reply:
I think if we change `dev` it also affects `cargo check`, but not sure. Also what about `rust-analyzer`?

Reply:
I'm assuming everything that builds uses `dev` unless `--release` is supplied.